ubccr/xdmod
# Fix Query Descriptor Visibility

**NOTE: I apologize in advance for the large number of files in this PR. On the plus side, the vast majority of them are test artifact updates.**

## Description
Updated to allow query descriptors to control visibility

**acl-config**
Previously we only allowed statistics to determine if an `acl_group_by` record
was visible or not. But our config files seem to indicate that certain query
descriptors should be able to control their ability to be shown as well. To
support both of these use cases the `processQueryDescriptors` function in
`acl-config` has been updated to take into account the `hide` property defined
at the query descriptor level and the `visible` property defined at the
statistic level. The logic used to determine the final visibility of an
`acl_group_by` record is as follows.

- By default all entries are shown
- If `visible` is defined then we use it's value.
- Else, if `hide` is defined then we use it's value.

Also, there was a minor refactor with the removal of the `$enabled` variable.
The `$disable` variable that it was based on was never going to be null, so it's
use could be replaced w/ just `!$disable`.

**Acls**
There was some minor cleanup done because it was determined that we always want
to have the `visible` column included, not just when a `$statisticName` is
provided.

This allowed us to move the `$descripter->setShowMenu(...)` from out of the
`if(isset($statisticName)) { ... }` guard so that all query descriptors
generated by this function will have their `showMenu` property populated.

These two sets of changes work together to allow us to control our UI state as
we expect to.

## Motivation and Context
Being able to control the visibility of the group bys / statistics via ACLs is expected. This was specifically a problem on an XSEDE install due to the more complicated visibility requirements. In OpenXDMoD there are no group by's / statistics that are hidden via the ACLs so this was not caught there. 

## Tests performed
Manual Testing was performed:
  - Navigate to an XSEDE Dev install, for each of the following users perform the steps below: pub, usr, cc
  - Select the `Usage` Tab. 
  - Ensure that the realms / group by's that are shown match user is allowed to see ( `etc/roles.d/pub.json#query_descripters`
      - Users and which entries should be hidden: 
          - *pub*: Accounts
          - *usr*: Accounts, Allocations
          - *cc*: Accounts, Allocations

## Types of changes


## Checklist:
